src/librustdoc/clean/mod.rs

```diff
@@ -2112,6 +2116,33 @@ impl Clean<Vec<Item>> for doctree::Impl<'_> {
             })
             .unwrap_or_default();
 
+        let for_ = self.for_.clean(cx);
+        let type_alias = for_.def_id().and_then(|did| match cx.tcx.def_kind(did) {
+            Some(DefKind::TyAlias) => Some(cx.tcx.type_of(did).clean(cx)),
+            _ => None,
+        });
+        if let Some(type_alias) = type_alias {
```

Comment:
The above match seems redundant. 

```suggestion
        if let Some(DefKind::TyAlias) = for_.def_id().and_then(|did| match cx.tcx.def_kind(did) {
        let type_alias = cx.tcx.type_of(did).clean(cx);
```

Reply:
If we do it this way, we'll have to clone `trait_`  and `items` one extra time. I'll write something else instead to avoid the duplication of code.